test/tests/classTransformer.test.ts

```diff
@@ -0,0 +1,118 @@
+import { ObjectId } from 'bson';
+import { classToPlain, Exclude, Expose, plainToClass, Transform } from 'class-transformer';
+
+import { prop } from '../../src/prop';
+import { getModelForClass } from '../../src/typegoose';
+
+// re-implement base Document to allow class-transformer to serialize/deserialize its properties
+class Document {
+  @Expose()
+  @Transform(
+    // makes sure that when deserializing from a Mongoose Object, ObjectId is serialized into a string 
+    (value: any) => value instanceof ObjectId
+    ? value.toHexString()
+    : value,
+    { toClassOnly: true })
+  public _id: string
+  
+  @Expose()
+  public __v: number
+}
+
+@Exclude()
+class Account extends Document {
+  @prop()
+  @Expose()
+  public email: string
+
+  @prop()
+  @Expose({ groups: ['admin'] })
+  password: string;
+}
+
+const AccountModel = getModelForClass(Account)
+
+describe('class-transformer', () => {
+  let id: string
+  beforeAll(async () => {
+    const { _id } = await AccountModel.create({
+      email: 'somebody@gmail.com',
+      password: 'secret',
+    } as Account)
+    // note here that _id is an ObjectId, hence the toString()
+    // otherwise it will have the shape of : { _bsonType: 'ObjectId', id: ArrayBuffer }
+    id = _id.toString()
+  })
+  describe('lean', () => {
+    it('should be able to use class-transformer', async () => {
+      // lean return a Plain Old Javascript Object
+      const POJO = await AccountModel.findById(id).lean()
+      // deserialize Plain Old Javascript Object into an instance of the Account class
```

Comment:
is this meant as in `POJO->Class(non-typegoose)` or `POJO->Class(typegoose)`?

Reply:
This is like `POJO->Class(non-typegoose)`. This is why there's a transformation round-trip (`plainToClass` then `classToPlain`) : since Typegoose class is not annotated with `class-transformer` decorators, it first requires to be turned into `Account`, then it has the decorators and can benefit from `class-transformer`'s transformations.

Reply:
Actually if using the annotated `Account` right away from the output of `typegoose`, there would be no need to do a `plainToClass` first. This is redundant, but required in this simple example.

Reply:
If I'm correct, this is because `typegoose` return a `DocumentType<Account>` while the decorators are on `Account`.

Reply:
yes, its thanks to mongoose, the return type is an instance of document (and model?) but not of the class

Reply:
Yes I think this is because of the nature of NoSQL, since it can store virtually any data structure in a collection.